### PR TITLE
feat: make portal-client dependency optional

### LINKED ISSUE #474: Make portal-client dependency optional
### User story

In order to streamline approval process in enterprises which vet every NPM package, it helps if the Code PushUp CLI has as few external dependencies as possible.

The `@code-pushup/portal-client` (which also pulls in `graphql`, `graphql-request` and `graphql-tag`) is only needed for uploading to portal. That means its installation could be optional.

### Acceptance criteria

- [ ] change `@code-pushup/portal-client` imports from static to dynamic
- [ ] move `@code-pushup/portal-client` from `dependencies` to `optionalDependencies`
- [ ] add additional portal setup instructions to documentation

### PR DESCRIPTION
**Pull request by hanna-skryl**
Closes #474 

- Make `@code-pushup/portal-client` an optional peer dependency in the `@code-pushup/core` package.
- Update the core package to use dynamic imports for `@code-pushup/portal-client`.
- Add a utility function to handle the loading of the portal client.
- Update the installation instructions to guide users on omitting optional dependencies, including details for `npm`, `pnpm`, and `yarn` setups.

### CODE REVIEW

File: e2e/plugin-lighthouse-e2e/tests/collect.e2e.test.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ describe('collect report with lighthouse-plugin NPM package', () => {
 
     const { code, stdout } = await executeProcess({
       command: 'npx',
-      args: ['@code-pushup/cli', 'collect', '--no-progress'],
+      args: ['@code-pushup/cli', 'collect', '--no-progress', '--verbose'],
```
Comment (BioPhoton) [inline e2e/plugin-lighthouse-e2e/tests/collect.e2e.test.ts:33]: Is this a left over?

Comment (hanna-skryl) [reply e2e/plugin-lighthouse-e2e/tests/collect.e2e.test.ts:33]: Yes, sorry for not providing the context for this change. I missed that the e2e test started failing in the previous PR that hid audits with perfect scores behind the `verbose` flag. 

Without `verbose`, this test was passing as is for Ubuntu and macOS, but for Windows, it was failing as the output there would be "All 4 audits have perfect scores". I'm not sure where this discrepancy is coming from.

Comment (BioPhoton) [reply e2e/plugin-lighthouse-e2e/tests/collect.e2e.test.ts:33]: Hmm... We should at least document this behaviour. 
Also I guess we should put a comment there IMO.

File: packages/cli/README.md [hunk 2]
```diff
@@ -44,6 +44,31 @@ _If you're looking for programmatic usage, then refer to the underlying [@code-p
...
+
+   ```sh
+   pnpm add --save-dev @code-pushup/cli
+   rm -rf pnpm-lock.yaml node_modules
+   pnpm install --no-optional
+   ```
+
+   **yarn**
+
+   `yarn` does not support omitting optional dependencies.
```
Comment (matejchalk) [inline packages/cli/README.md:70]: Wouldn't [`--ignore-optional`](https://classic.yarnpkg.com/lang/en/docs/cli/install/#toc-yarn-install-ignore-optional) work? :thinking:

**Review by matejchalk — APPROVED** [review]

File: e2e/plugin-lighthouse-e2e/tests/collect.e2e.test.ts (hunk 1 above)
Comment (hanna-skryl) [reply e2e/plugin-lighthouse-e2e/tests/collect.e2e.test.ts:34]: Added a comment explaining why the test must be run in `verbose` mode.

**Review by BioPhoton — APPROVED** [review]

**Pull request merged**